### ISSUE TITLE
[QST] Converting graph_view_t to legacy::GraphCSRView

### ISSUE THREAD

**Issue by wbthomason** [rapidsai/cugraph#2372]
Hi! I'm trying to use `cugraph::connected_components` with `libcugraph`. I have a `cugraph::Graph` and corresponding `cugraph::graph_view_t` created from my data, which is in edgelist form. However, the `cugraph::connected_components` routine seems to only accept a `legacy::GraphCSRView`, which looks like it needs to be constructed manually. I wanted to see if I was missing something (i.e. an easy conversion between `graph_view_t` and `GraphCSRView`) - is there a recommended way to do this conversion, or should I just compute an offsets list manually? Thanks!

**Comment by seunghwak** [comment]
Do you need to compute strongly connected components or weakly connected components will work?

We have a weakly connected components implementation that will work with graph_veiw_t.

https://github.com/rapidsai/cugraph/blob/branch-22.08/cpp/include/cugraph/algorithms.hpp#L1427

**Comment by ChuckHastings** [comment]
Agreed.  If you only use WCC then the link Seunghwa provided is a version that handles that version.

If you need strongly connected components, we don't currently have an SCC implementation for the new graph object.  In that case you would need to construct `legacy::GraphCSRView` as you have observed.  Please note that the legacy graph objects *only* work on a single GPU.

There are two reasonable approaches for that:
1. If you still have the original data, you can create a `legacy::GraphCSR` with the original data, then create a `legacy::GraphCSRView` from it.
2. If you do not have the original data, you can use https://github.com/rapidsai/cugraph/blob/fa701f5f223093bb0d7f0e721efce10326ac05f4/cpp/include/cugraph/graph_view.hpp#L1045 to extract the data and then create the `legacy::GraphCSR` and then a `legacy::GraphCSRView` from that.

**Comment by wbthomason** [comment]
Thanks for the responses! I unfortunately do need strongly connected components (I'm looking at reachability within a digraph for path planning purposes, so a WCC, which ignores edge directions, may be misleading). I'm in a strict single-GPU domain, so the legacy object requirement isn't a burden - the problem is more running out of vRAM, as I'm operating on large (multi-GB) graphs and need to perform both SSSP and SCC on the same graph.

From what you've said, it sounds like my best bet is to do all my SSSP operations first, then unload the `graph_view_t` representation and construct a new `legact::GraphCSR` and `legacy::GraphCSRView` for SCC afterward?

**Comment by ChuckHastings** [comment]
YW!

I suggest tracking this issue.  Once complete you should be able to do all of your work with the new graph objects once this is addressed.  https://github.com/rapidsai/cugraph/issues/2144
